vllm/engine/llm_engine.py

```diff
@@ -845,7 +854,33 @@ def _process_model_outputs(
         for seq_group in ignored_seq_groups:
             request_output = RequestOutputFactory.create(seq_group)
             request_outputs.append(request_output)
-        return request_outputs
+        self.request_outputs = request_outputs
+        return
+
+    def _advance_to_next_step(
+        self,
+        output: List[SamplerOutput],
+        seq_group_metadata_list: List[SequenceGroupMetadata]) -> None:
+        """Given model output from a single run, append the tokens to the
+        sequences. This is normally done inside output processor, but it is
+        required if the worker is to perform async forward pass to next step.
+        """
+        for seq_group_metadata, sequence_group_outputs in zip(
+                seq_group_metadata_list, output):
```

Comment:
As we discussed, most likely you need to do seq.is_finished() check here before doing append_token(..)

Reply:
Updated now.